app/assets/javascripts/search.js

```diff
@@ -18,9 +20,71 @@ $(document).ready(function() {
     termsInput.prop("disabled", true);
     return true;
   });
-  $(".search-terms").keyup(function(e) {
+
+  var clickOnEnter = function(e, cls) {
     if (e.keyCode === 13) {
-      $(".m-search-ui button.search").trigger("click");
+      $(cls).trigger("click");
     }
+  };
+  $(".search-terms").keyup(function(e) {
+    clickOnEnter(e, ".m-search-ui button.search");
+  });
+
+  /* *** setup Save as Report *** */
+  $("#save-search form input").keyup(function(e) {
+    clickOnEnter(e, "#save-search-button");
+  });
+
+  // defined inline on HTML page
+  if (typeof C2searchQuery != "undefined") {
+    $("#save-search-query").text(C2searchQuery.humanized);
+  }
+  $("#save-search form").on("submit", function(e) {
+    e.preventDefault();
+  });
+  $("#save-search-button").click(function() {
+    var btn = $(this);
+    var form = $("#save-search form");
```

Comment:
I like the explicit naming of `savedSearchName` below -- perhaps this var would be more clear if it were named `savedSearchForm` ?